node/finder.ts

```diff
@@ -4,8 +4,26 @@ import { basename, extname, join } from 'path'
 import { EdgeFunction } from './edge_function.js'
 import { nonNullable } from './utils/non_nullable.js'
 
+// the order of the allowed extensions is also the order we remove duplicates
+// with a lower index meaning a higher precedence over the others
 const ALLOWED_EXTENSIONS = new Set(['.js', '.jsx', '.ts', '.tsx'])
```

Comment:
Sets don't (theoretically) hold order, so if order now matters should this become an array?